### PR TITLE
[SW-951] Add gripper open and close services to spot_ros2

### PR DESCRIPTION
**Pull request by tcappellari-bdai**
## Change Overview

Added services `open_gripper` and `close_gripper` to `spot_ros2`

## Testing Done

Please create a checklist of tests you plan to do and check off the ones that have been completed successfully. Ensure that ROS 2 tests use `domain_coordinator` to prevent port conflicts. Further guidance for testing can be found on the [ros utilities wiki](https://github.com/bdaiinstitute/ros_utilities/wiki/Testing-guidelines).

### CODE REVIEW

**Comment by tcappellari-bdai** [comment]
* **#344** <a href="https://app.graphite.dev/github/pr/bdaiinstitute/spot_ros2/344?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tcappellari-bdai and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by bhung-bdai — APPROVED** [review]

**Comment by khughes-bdai** [comment]
Can you also make this conditional on if the spot has an arm?

**Comment by tcappellari-bdai** [comment]
> Can you also make this conditional on if the spot has an arm?

it is, its inside the same if statement as the arm ones

**Comment by khughes-bdai** [comment]
Oops sorry graphite wasn't showing me the lines above

**Review by khughes-bdai — APPROVED** [review]
LGTM

File: spot_driver/spot_driver/spot_ros2.py [hunk 1]
```diff
@@ -1309,6 +1326,24 @@ def handle_carry(self, request: Trigger.Request, response: Trigger.Response) ->
...
+        return response
+
+    def handle_close_gripper(self, request: Trigger.Request, response: Trigger.Response) -> Trigger.Response:
+        """ROS service handler to close the gripper."""
+        if self.spot_wrapper is None:
+            response.success = False
+            response.message = "Spot wrapper is undefined"
+            return response
+        response.success, response.message = self.spot_wrapper.spot_arm.gripper_close()
+        return response
```
Comment (amessing-bdai) [inline spot_driver/spot_driver/spot_ros2.py:1345]: Should this be a decorator?

Comment (tcappellari-bdai) [reply spot_driver/spot_driver/spot_ros2.py:1345]: ya i was having the same thought as i copy and pasted more trigger services into this file - ill make a ticket

**Pull request merged**